docker/src/test/java/org/jclouds/docker/features/NetworkApiLiveTest.java

```diff
@@ -0,0 +1,123 @@
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.common.base.Predicates;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+
+@Test(groups = "live", testName = "NetworkApiLiveTest", singleThreaded = true)
+public class NetworkApiLiveTest extends BaseDockerApiLiveTest {
+
+   private static final String NETWORK_NAME = "JCLOUDS_NETWORK";
+
+   private Network network = null;
+   protected Image image = null;
+   private Container container;
+
+   @BeforeClass
+   protected void init() {
+
+      if (api.getImageApi().inspectImage(ALPINE_IMAGE_TAG) == null) {
+         CreateImageOptions options = CreateImageOptions.Builder.fromImage(ALPINE_IMAGE_TAG);
+         InputStream createImageStream = api.getImageApi().createImage(options);
+         consumeStream(createImageStream);
+      }
+      image = api.getImageApi().inspectImage(ALPINE_IMAGE_TAG);
+      assertNotNull(image);
+
+      Config containerConfig = Config.builder().image(image.id())
+              .cmd(ImmutableList.of("sh", "-c", "touch hello; while true; do echo hello world; sleep 1; done"))
+              .build();
+      container = api.getContainerApi().createContainer("jclouds-test-network", containerConfig);
+      api.getContainerApi().startContainer(container.id());
+      container = api.getContainerApi().inspectContainer(container.id());
+   }
+
+   @AfterClass
+   protected void tearDown() {
+      if (container != null) {
+         api.getContainerApi().stopContainer(container.id());
+         api.getContainerApi().removeContainer(container.id());
+      }
+      if (network != null) {
+         api().removeNetwork(network.id());
+      }
+   }
+
+   public void testCreateNetwork() throws IOException, InterruptedException {
+      network = api().createNetwork(Network.create(NETWORK_NAME, null, null, null, null, ImmutableMap.<String, Network.Details> of(), ImmutableMap.<String, String> of()));
+      assertNotNull(network);
+      assertNotNull(network.id());
+
+   }
+
+   @Test(dependsOnMethods = "testCreateNetwork")
+   public void testGetNetwork() {
+      network = api().inspectNetwork(network.id());
+      assertNotNull(network);
+   }
+
+   @Test(dependsOnMethods = "testGetNetwork")
+   public void testAttachContainerToNetwork() {
+      api().connectContainerToNetwork(network.id(), container.id());
+      Container foundContainer = api.getContainerApi().inspectContainer(container.id());
+      assertNotNull(Iterables.tryFind(foundContainer.networkSettings().networks().keySet(), Predicates.equalTo(network.name())).orNull());
+   }
+
+   @Test(dependsOnMethods = "testAttachContainerToNetwork")
+   public void testDisconnectContainerFromNetwork() {
+      api().disconnectContainerFromNetwork(network.id(), container.id());
+      // TODO assertion
+   }
+
+   @Test(dependsOnMethods = "testDisconnectContainerFromNetwork")
```

Comment:
Make it depend on the `testCreateNetwork` to maximise the chances it is not skipped due to a previous test failure.